eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/client/EclairRpcClient.scala

```diff
@@ -568,4 +667,47 @@ class EclairRpcClient(val instance: EclairInstance)(
   def stop(): Option[Unit] = {
     process.map(_.destroy())
   }
+
+  /**
+    * Pings eclair every second to see if a invoice has been paid
+    * If the invoice has been paid or the payment has failed, we publish a
+    * [[org.bitcoins.eclair.rpc.json.PaymentResult PaymentResult]]
+    * event to the [[akka.actor.ActorSystem ActorSystem]]'s
+    * [[akka.event.EventStream ActorSystem.eventStream]]
+    */
+  override def monitorSentPayment(
+      paymentId: PaymentId, notifyPayment: PaymentResult => Future[Unit] = notifySentPayment): Future[PaymentResult] = {
+    val p: Promise[PaymentResult] = Promise[PaymentResult]()
+
+    val runnable = new Runnable() {
+
+      private var attempt = new AtomicInteger(1)
+
+      override def run(): Unit = {
+        for {
+          results <- getSentInfo(paymentId)
+        } yield {
+          results.find(_.status != PaymentStatus.PENDING) match {
```

Comment:
Why not `results.find(_.status == PaymentStatus.SUCCEEDED)`?

Reply:
Because the payment can fail.

Reply:
So here is the documentation from eclair

https://acinq.github.io/eclair/

```
Returns a list of attempts to send an outgoing payment, the possible statuses are PENDING, FAILED and SUCCEEDED. The API can be queried by paymentHash OR by uuid, if the latter is used the API returns a list containing at most one element.
```
That means this implementation will find the _first_ attempt at sending the payment, not the last one. I guess this opens a question here of which one we are interested in. It seems we want to search for a `SUCCESS` until the timeout occurrs? I still am not convinced that this is the right solution